### PR TITLE
This is the code and testing

### PR DESCRIPTION
**Pull request by shubhadapaithankar**
**This PR is divided in to two parts** 

### Part 1:
This code makes it easier to test and use the Greet function in different places by allowing us to control where messages go.

Error:
```
% go test
--- FAIL: TestGreet (0.00s)
    di_test.go:16: got "Hello, Chris" want "Hello,Chris"
FAIL
exit status 1
FAIL    test/hello/Dependancy-Injection 0.203s
```

**Fixed :** The fixed was the expected output in the test for the Greet function, updating it to include a space after the comma ("Hello, Chris") to match the actual output and standard punctuation practices.

```
% go test
PASS
ok      test/hello/Dependancy-Injection 0.313s
```

**Part 2:**
Mocking: This part of PR introduces mocking to enhance test suite's isolation and speed by simulating external dependencies, ensuring focused and deterministic tests without relying on real system components.

### CODE REVIEW

File: Dependancy-Injection/di.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+// Greet sends a greeting to a writer.
+func Greet(writer io.Writer, name string) {
+	// fmt.Fprintf to format the greeting message and write it to the provided writer.
+	//fmt.Fprintf(writer, "Hello, %s", name)
```
Comment (SudoBrendan) [inline Dependancy-Injection/di.go]: nit: dead code should be removed instead of commented out - we'll have git history if we want to revisit later.

**Review by SudoBrendan — APPROVED** [review]
The code is great! Ultimately LGTM, I'd be good to merge.

nit-pick (I typically shorten this to `nit`...): the comments you have above `func` calls are very useful because they give us [godoc](https://pkg.go.dev/golang.org/x/tools/cmd/godoc), but I don't think comments other than those lift much in this change.

Removing those comments gives us two benefits:

1. it's 1/2 as many lines to review
2. it's 1/2 as many lines to update later if we change the code, and eliminates any divergence between comments and code that might show up later because someone forgets to update the comment

In short: **your code is elegant and straightforward**, even without the comments - I'd let your code speak for itself :)
